portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -135,18 +149,27 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
 
         if (id > 0) {
             Key commentEntityKey = KeyFactory.createKey("Comment", id);
-            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-            datastore.delete(commentEntityKey);
+
+            Query.Filter queryFilter = new Query.FilterPredicate("__key__", Query.FilterOperator.EQUAL, commentEntityKey);
+            query.setFilter(queryFilter);
+            PreparedQuery results = datastore.prepare(query);
+            Entity entity = results.asSingleEntity();
+
+            String commentUserId = (String) entity.getProperty("userId");
+            
+
+            // Only allows comments written by user logged in to be deleted
+            if (userId.equals(commentUserId)){
```

Comment:
Likewise, we can make datastore do this work by handing it two filters with `CompositeFilter`: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Query.CompositeFilter?hl=en#class-query.compositefilter.

Reply:
Thanks! I hadn't read up on `CompositeFilter` before. This'll work better!